### PR TITLE
[SPoG] Add guac integration

### PR DESCRIPTION
**Pull request by dejanb**
This is continuation of #383

### CODE REVIEW

File: deploy/compose/compose-trustification.yaml [hunk 1]
```diff
@@ -43,37 +43,37 @@ services:
...
+  #     # the issuer URL is passed to the browser, which is running on the host, not insider the container, so the
+  #     # URL is "localhost"
+  #     - ISSUER_URL=http://localhost:8090/realms/chicken
+  #   depends_on:
+  #     - spog-api
+  #   restart: on-failure
+  #   expose:
+  #     - "$SPOG_UI_PORT"
+  #   ports:
+  #     - "$SPOG_UI_PORT:8080"
```
Comment (lulf) [inline deploy/compose/compose-trustification.yaml]: You probably meant to revert this.

File: infrastructure/src/infra.rs [hunk 2]
```diff
@@ -13,7 +13,7 @@ use crate::tracing::init_tracing;
 #[cfg(unix)]
 use tokio::signal::unix::{signal, SignalKind};
 
-const DEFAULT_BIND_ADDR: &str = "localhost:9010";
+const DEFAULT_BIND_ADDR: &str = "[::1]:9010";
```
Comment (lulf) [inline infrastructure/src/infra.rs:16]: Whats the reason for changing this? I'm worried about ipv6 vs ipv4 issues.

File: spog/api/src/lib.rs [hunk 3]
```diff
@@ -22,20 +24,17 @@ pub const DEFAULT_CRDA_PAYLOAD_LIMIT: usize = 10 * 1024 * 1024;
...
 pub struct Run {
-    #[command(flatten)]
-    pub snyk: Snyk,
-
     /// Enable developer mode
     #[arg(long = "devmode", default_value_t = false)]
     pub devmode: bool,
 
-    #[arg(short, long, env, default_value = "0.0.0.0")]
+    #[arg(short, long, env, default_value = "::")]
```
Comment (lulf) [inline spog/api/src/lib.rs:31]: Same ipv4 vs ipv6 worries here :)

File: spog/api/src/services/mod.rs [hunk 4]
```diff
@@ -0,0 +1 @@
+pub mod guac;
```
Comment (lulf) [inline spog/api/src/services/mod.rs:1]: Probably pub(crate) is sufficient.

File: spog/api/src/lib.rs [hunk 5]
```diff
@@ -1,12 +1,14 @@
 use std::process::ExitCode;
 use std::{net::TcpListener, path::PathBuf};
-use trustification_auth::authenticator::config::AuthenticatorConfig;
-use trustification_auth::swagger_ui::SwaggerUiOidcConfig;
+use trustification_auth::{authenticator::config::AuthenticatorConfig, swagger_ui::SwaggerUiOidcConfig};
 use trustification_infrastructure::{Infrastructure, InfrastructureConfig};
 
+pub mod services;
```
Comment (lulf) [inline spog/api/src/lib.rs:6]: Probably pub(crate) is sufficient.

**Review by lulf — COMMENTED** [review]
Left some comments on network bind and module visibility. Also:

* I would just drop the additional services module and align guac with other services like bombastic or vexination in the module structure.
* I think it would be good to add the guac response data structures, even if it's just a Vec<String>, to the spog/model crate so that the client can be kept in sync with the api.

File: infrastructure/src/infra.rs (hunk 2 above)
Comment (dejanb) [reply infrastructure/src/infra.rs:16]: Already reverted :)

**Comment by dejanb** [comment]
@lulf finally it's all green. One final look before merging?

**Pull request merged**